apps/system/utils/fscmd.c

```diff
@@ -993,6 +993,55 @@ static int tash_pwd(int argc, char **args)
 }
 #endif
 #ifndef CONFIG_DISABLE_ENVIRON
+static int delete_entry(const char *fullpath)
+{
+	int ret = ERROR;
+	struct stat st;
+	if (fullpath) {
+		if (stat(fullpath, &st) < 0) {
+			FSCMD_OUTPUT("stat failed with %s\n", fullpath);
+		} else if (!S_ISDIR(st.st_mode)) {
+			ret = unlink(fullpath);
+			if (ret == OK) {
+				FSCMD_OUTPUT("%s deleted\n", fullpath);
+			}
+		} else {
+			/* Iterate the directory contents */
+			DIR *dirp = opendir(fullpath);		/* Open the directory */
+			if (!dirp) {
+				/* Failed to open the directory */
+				FSCMD_OUTPUT("\t Failed to open directory: %s\n", fullpath);
+				return ERROR;
+			}
+
+			/* Read each directory entry */
+			for (;;) {
+				char *entrypath = NULL;
+				struct dirent *entryp = readdir(dirp);
+				if (!entryp) {
+					/* Finished with this directory */
+					break;
+				}
+
+				/* Call delete_entey recursively */
+				entrypath = get_dirpath(fullpath, entryp->d_name);
+				delete_entry(entrypath);
```

Comment:
Please add return of delete_entry and break if returns error.

Reply:
Right. return is added.